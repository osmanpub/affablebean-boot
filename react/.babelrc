 
{
  "presets": ["@babel/preset-react"],
  "env": {
    "test": {
      "plugins": [
        ["istanbul", {
          "exclude": [
            "**/*.spec.js"
          ]
        }]
      ]
    }
  },
  "plugins": ["transform-class-properties", "istanbul"]
}
