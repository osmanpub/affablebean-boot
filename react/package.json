{
  "name": "affablebean",
  "version": "0.1.0",
  "private": true,
  "eslintConfig": {
    "extends": "react-app",
    "plugins": [
      "react-hooks"
    ],
    "rules": {
      "react-hooks/rules-of-hooks": "error",
      "react-hooks/exhaustive-deps": "warn"
    }
  },
  "scripts": {
    "build": "react-scripts build",
    "coveralls": "cat coverage/lcov.info | coveralls",
    "cypress:open": "cypress open",
    "cypress:run": "cypress run",
    "cypress:verify": "cypress verify",
    "eject": "react-scripts eject",
    "report:coverage": "nyc report --reporter=html",
    "report:coverage:text": "nyc report --reporter=text",
    "start": "node_modules/react-scripts/bin/react-scripts.js start &",
    "start:java": "cd .. && sh server && cd react && node_modules/react-scripts/bin/react-scripts.js start &",
    "test": "react-scripts test"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "dependencies": {
    "@reduxjs/toolkit": "^1.2.5",
    "axios": "^0.19.2",
    "node-rest-client": "^3.1.0",
    "react": "^16.12.0",
    "react-alert": "^6.0.0",
    "react-alert-template-basic": "^1.0.0",
    "react-dom": "^16.12.0",
    "react-hook-form": "^4.9.7",
    "react-redux": "^7.2.0",
    "react-router-dom": "^5.1.2",
    "react-scripts": "3.4.0",
    "redux": "^4.0.5",
    "styled-components": "^5.0.1"
  },
  "devDependencies": {
    "@babel/core": "^7.8.4",
    "@babel/preset-react": "^7.8.3",
    "@cypress/code-coverage": "^1.12.2",
    "@testing-library/jest-dom": "^4.2.4",
    "@testing-library/react": "^9.4.0",
    "@testing-library/user-event": "^7.2.1",
    "@types/axios": "^0.14.0",
    "@types/cypress": "^1.1.3",
    "@types/jest": "^25.1.3",
    "@types/react": "^16.9.21",
    "@types/react-alert": "^5.2.0",
    "@types/react-bootstrap": "^0.32.20",
    "@types/react-dom": "^16.9.5",
    "@types/react-redux": "^7.1.7",
    "@types/react-router-dom": "^5.1.3",
    "@types/react-test-renderer": "^16.9.2",
    "@types/redux": "^3.6.0",
    "@types/redux-mock-store": "^1.0.2",
    "@types/styled-components": "^5.0.0",
    "babel-plugin-istanbul": "^6.0.0",
    "babel-plugin-transform-class-properties": "^6.24.1",
    "coveralls": "^3.0.9",
    "cross-env": "^7.0.0",
    "cypress": "^4.0.2",
    "istanbul-lib-coverage": "^3.0.0",
    "nyc": "^15.0.0",
    "react-test-renderer": "^16.12.0",
    "redux-mock-store": "^1.5.4",
    "typescript": "^3.8.2"
  },
  "nyc": {
    "sourceMap": false,
    "instrument": false
  }
}
