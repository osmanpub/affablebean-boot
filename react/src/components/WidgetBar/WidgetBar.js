import React from "react";
import { Link } from "react-router-dom";
import { Checkout, ViewCart, WidgetBarWrapper } from "./WidgetBar.styles";
import "./WidgetBar.css";

export function WidgetBar(props) {
  const cart = props.cart;

  return (
    <WidgetBarWrapper>
      <Checkout className="headerWidget" {...cart}>
        <div>
          <Link className="bubble" to={"/checkout"}>
            proceed to checkout &#x279f;
          </Link>
        </div>
      </Checkout>

      <ViewCart className="headerWidget" {...cart}>
        <span className="horizontalMargin">
          <img src="/static/img/cart.gif" alt="shopping cart icon" />
          &nbsp;
          {cart && cart.numberOfItems + " items"}
        </span>

        <Link className="bubble" to={"/viewCart/false"}>
          view cart
        </Link>
      </ViewCart>
    </WidgetBarWrapper>
  );
}
