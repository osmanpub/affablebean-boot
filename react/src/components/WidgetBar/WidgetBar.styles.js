import styled from "styled-components";

export const WidgetBarWrapper = styled.div`
  height: 50px;
  width: 1080px;
  float: right;
`;

export const Checkout = styled.div`
  display: ${cart => (cart.numberOfItems > 0 ? "block" : "none")};
`;

export const ViewCart = styled.div`
  display: ${cart => (cart.numberOfItems > 0 ? "block" : "none")};
`;

// #logo {
// 	float: left;
// 	margin-left: 30px;
// 	margin-top: -20px;
// }

// #cart {
// 	position: absolute;
// 	margin: 5px 0 0 2px;
// }

// #viewCart {
// 	text-align: left;
// 	width: 210px;
// }
