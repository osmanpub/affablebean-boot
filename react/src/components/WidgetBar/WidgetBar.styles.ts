import styled from "styled-components";
import { Cart } from "../../interfaces/cart";

export const WidgetBarWrapper = styled.div`
  height: 50px;
  width: 1080px;
  float: right;
`;

export const Checkout = styled.div`
  display: ${(cart: Cart) => (cart.numberOfItems > 0 ? "block" : "none")};
`;

export const ViewCart = styled.div`
  display: ${(cart: Cart) => (cart.numberOfItems > 0 ? "block" : "none")};
`;
