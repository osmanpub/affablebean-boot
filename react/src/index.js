import React from "react";
import { render } from "react-dom";
import { configureStore } from "redux-starter-kit";
import { Provider } from "react-redux";
import { BrowserRouter as Router, Route } from "react-router-dom";
import reducer from "./reducers";
import App from "./containers/App";
import CategoryProducts from "./containers/CategoryProducts";
import * as serviceWorker from "./serviceWorker";

const store = configureStore({
  reducer
});

render(
  // <React.StrictMode>
  <Provider store={store}>
    <Router>
      <Route path="/" exact component={App} />
      <Route path="/category/:id" component={CategoryProducts} />
    </Router>
  </Provider>,
  // </React.StrictMode>,
  document.getElementById("root")
);

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: http://bit.ly/CRA-PWA
serviceWorker.unregister();
