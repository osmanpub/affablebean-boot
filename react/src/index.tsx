import { configureStore } from "@reduxjs/toolkit";
import React, { lazy, Suspense } from "react";
import { render } from "react-dom";
import { Provider } from "react-redux";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import ErrorBoundary from "./components/ErrorBoundary";
import rootReducer from "./redux";
import * as serviceWorker from "./serviceWorker";
const Cart = lazy(() => import("./containers/Cart"));
const CategoryProducts = lazy(() => import("./containers/CategoryProducts"));
const Checkout = lazy(() => import("./containers/Checkout"));
const Contact = lazy(() => import("./containers/Contact"));
const Home = lazy(() => import("./containers/Home"));
const Privacy = lazy(() => import("./containers/Privacy"));

const store = configureStore({
  reducer: rootReducer
});

render(
  <React.StrictMode>
    <ErrorBoundary>
      <Provider store={store}>
        <Router>
          <Suspense fallback={<div>Loading...</div>}>
            <Switch>
              <Route path="/" exact component={Home} />
              <Route path="/category/:id" component={CategoryProducts} />
              <Route path="/checkout" component={Checkout} />
              <Route path="/contact" component={Contact} />
              <Route path="/privacy" component={Privacy} />
              <Route path="/viewCart/:clear" component={Cart} />
            </Switch>
          </Suspense>
        </Router>
      </Provider>
    </ErrorBoundary>
  </React.StrictMode>,
  document.getElementById("root")
);

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: http://bit.ly/CRA-PWA
serviceWorker.unregister();
