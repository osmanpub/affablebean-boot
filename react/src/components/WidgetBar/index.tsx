import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { Cart } from "../../interfaces/cart";
import { RootState } from "../../redux";
import "./WidgetBar.css";
import { Checkout, ViewCart, WidgetBarWrapper } from "./WidgetBar.styles";

type Props = {
  cart?: Cart;
  url: string;
};

export function WidgetBar(props: Props) {
  const { cart, url } = props;
  const viewCart = url.startsWith("/viewCart") ? (
    ""
  ) : (
    <Link
      className="bubble"
      to={"/viewCart/false"}
      data-cy="cart-hdr-view-cart"
    >
      view cart
    </Link>
  );
  const checkout = url.startsWith("/checkout") ? (
    ""
  ) : (
    <Checkout className="headerWidget" {...cart}>
      <Link className="bubble" to={"/checkout"} data-cy="proceed-checkout">
        proceed to checkout &#x279f;
      </Link>
    </Checkout>
  );

  return (
    <WidgetBarWrapper>
      {checkout}

      <ViewCart className="headerWidget" {...cart}>
        <span className="horizontalMargin" data-cy="cart-hdr-total">
          <img src="/static/img/cart.gif" alt="shopping cart icon" />
          &nbsp;
          {cart && cart.numberOfItems + " items"}
        </span>
        {viewCart}
      </ViewCart>
    </WidgetBarWrapper>
  );
}

const mapStateToProps = (state: RootState) => ({
  cart: state.cart
});

export default connect(mapStateToProps)(WidgetBar);
