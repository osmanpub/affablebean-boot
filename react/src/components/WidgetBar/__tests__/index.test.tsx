import { render } from "@testing-library/react";
import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import WidgetBar from "..";

const mockStore = configureStore([]);

const initialState = {
  cart: {
    didInvalidate: false,
    isFetching: false,
    items: [
      {
        name: "butter",
        price: 1.09,
        description: "unsalted (250g)"
      }
    ],
    numberOfItems: 1,
    subtotal: 1.09
  }
};

const store = mockStore(initialState);

const widgetBar = (
  <Router>
    <Provider store={store}>
      <WidgetBar url="/" />
    </Provider>
    )
  </Router>
);

describe("<WidgetBar />", () => {
  it("renders correctly", () => {
    const component = renderer.create(widgetBar).toJSON();
    // @ts-ignore
    expect(component).toMatchSnapshot();
  });

  it("shows view cart link", () => {
    const { getByText } = render(widgetBar);
    const link = getByText(/view cart/i);
    // @ts-ignore
    expect(link).toBeInTheDocument();
  });

  it("shows proceed to checkout link", () => {
    const { getByText } = render(widgetBar);
    const link = getByText(/proceed to checkout/i);
    // @ts-ignore
    expect(link).toBeInTheDocument();
  });

  it("cart header total has 1 item", () => {
    const { getByText } = render(widgetBar);
    const cart = getByText(/1 items/i);
    // @ts-ignore
    expect(cart).toBeInTheDocument();
  });
});
