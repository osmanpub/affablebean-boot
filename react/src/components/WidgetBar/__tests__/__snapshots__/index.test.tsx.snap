// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WidgetBar /> renders correctly 1`] = `
Array [
  <div
    className="sc-AxjAm cHCNiJ"
  >
    <div
      className="sc-AxirZ fmVTfn headerWidget"
    >
      <a
        className="bubble"
        data-cy="proceed-checkout"
        href="/checkout"
        onClick={[Function]}
      >
        proceed to checkout ➟
      </a>
    </div>
    <div
      className="sc-AxiKw RnNev headerWidget"
    >
      <span
        className="horizontalMargin"
        data-cy="cart-hdr-total"
        data-testid="cart-hdr-total"
      >
        <img
          alt="shopping cart icon"
          src="/static/img/cart.gif"
        />
         
        1 items
      </span>
      <a
        className="bubble"
        data-cy="cart-hdr-view-cart"
        href="/viewCart/false"
        onClick={[Function]}
      >
        view cart
      </a>
    </div>
  </div>,
  ")",
]
`;
